webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationMessageSenders.java

```diff
@@ -0,0 +1,237 @@
+import com.box.l10n.mojito.service.branch.notification.BranchNotificationMessageSendersConfigurationProperties.GithubConfigurationProperties;
+import com.box.l10n.mojito.service.branch.notification.BranchNotificationMessageSendersConfigurationProperties.NoopConfigurationProperties;
+import com.box.l10n.mojito.service.branch.notification.BranchNotificationMessageSendersConfigurationProperties.PhabricatorConfigurationProperties;
+import com.box.l10n.mojito.service.branch.notification.BranchNotificationMessageSendersConfigurationProperties.SlackConfigurationProperties;
+import com.box.l10n.mojito.service.branch.notification.BranchNotificationMessageSendersConfigurationProperties.SlackConfigurationProperties.MessageBuilderConfigurationProperties;
+import com.box.l10n.mojito.service.branch.notification.github.BranchNotificationMessageSenderGithub;
+import com.box.l10n.mojito.service.branch.notification.noop.BranchNotificationMessageSenderNoop;
+import com.box.l10n.mojito.service.branch.notification.phabricator.BranchNotificationMessageBuilderPhabricator;
+import com.box.l10n.mojito.service.branch.notification.phabricator.BranchNotificationMessageSenderPhabricator;
+import com.box.l10n.mojito.service.branch.notification.slack.BranchNotificationMessageBuilderSlack;
+import com.box.l10n.mojito.service.branch.notification.slack.BranchNotificationMessageSenderSlack;
+import com.box.l10n.mojito.slack.SlackClient;
+import com.box.l10n.mojito.slack.SlackClients;
+import com.box.l10n.mojito.thirdpartynotification.slack.SlackChannels;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BranchNotificationMessageSenders {
+
+  final Map<String, BranchNotificationMessageSender> mapIdToBranchNotificationMessageSender;
+
+  public BranchNotificationMessageSenders(
+      BranchNotificationMessageSendersConfigurationProperties
+          branchNotificationMessageSendersConfigurationProperties,
+      GithubClients githubClients,
+      SlackClients slackClients,
+      BranchUrlBuilder branchUrlBuilder) {
+    this.mapIdToBranchNotificationMessageSender =
+        createInstancesFromConfiguration(
+            branchNotificationMessageSendersConfigurationProperties,
+            githubClients,
+            slackClients,
+            branchUrlBuilder);
+  }
+
+  public BranchNotificationMessageSender getById(String id) {
+    return mapIdToBranchNotificationMessageSender.get(id);
+  }
+
+  private Map<String, BranchNotificationMessageSender> createInstancesFromConfiguration(
+      BranchNotificationMessageSendersConfigurationProperties
+          branchNotificationMessageSendersConfigurationProperties,
+      GithubClients githubClients,
+      SlackClients slackClients,
+      BranchUrlBuilder branchUrlBuilder) {
+
+    checkIdsStartWithPrefix(
```

Comment:
Why is it important to validate the name of the keys?
Do we functionally rely on the naming pattern/prefix latter on to drive some logic/instantiate the right class? (that wouldn't be ideal if that's the case)

Or is this purely so as to enforce consistent naming in the config files?